doc/sphinx/language/extensions/compil-steps.rst

```diff
@@ -0,0 +1,80 @@
+.. _compil-steps:
+
+======================
+Main Compilation Steps
+======================
+
+It is in theory possible to write down every Gallina term explicitly
+as described in the :ref:`Core Language <core-language>` part of this
+manual, in the way they are understood by the kernel, for instance
+:n:`Nat.add (S O) (Nat.mul (S (S O)) (S (S O)))`. However, this would
```

Comment:
that is not the way the kernel sees them, kernel names are fully explicit, split names in (library name / module name / label) and constructors are (name of inductive / constructor index)
(and then there's the kerpair stuff)

Reply:
Sure, but do we want to enter into that much details here? We could write `Stdlib.Init.Nat.add (Stdlib.Init.Datatypes.S Stdlib.Init.Datatypes.O) (Stdlib.Init.Nat.mul (Stdlib.Init.Datatypes.S (Stdlib.Init.Datatypes.S Stdlib.Init.Datatypes.O)) (Stdlib.Init.Datatypes.S (Stdlib.Init.Datatypes.S Stdlib.Init.Datatypes.O)))` as an intermediate solution, but that's already pretty verbose.